### PR TITLE
overlord/servicestate: disallow removal of quota group with any limits set

### PR DESCRIPTION
**Pull request by Meulengracht**
Missing case when removing a quota group with a limit set.

### CODE REVIEW

File: snap/quota/resources.go [hunk 1]
```diff
@@ -203,14 +203,19 @@ func (qr *Resources) CheckFeatureRequirements() error {
 	return nil
 }
 
+// IsZero returns true if there is no limit set
+func (qr *Resources) IsZero() bool {
```
Comment (pedronis) [inline snap/quota/resources.go:207]: Unset() bool might be preferable here?

Comment (Meulengracht) [reply snap/quota/resources.go:207]: `Unset` implies action, I'd rather have IsUnset to clarify a boolean result - IsZero is used multiple places by golang.

Comment (Meulengracht) [reply snap/quota/resources.go:207]: I'll change it to Unset so it matches our current convention, but my opinion is it's not logical for that operation.

And thanks for the review!

File: snap/quota/resources.go [hunk 2]
```diff
@@ -203,14 +203,19 @@ func (qr *Resources) CheckFeatureRequirements() error {
 	return nil
 }
 
+// Unset returns true if there is no limit set
+func (qr *Resources) Unset() bool {
+	return qr.Memory == nil && qr.CPU == nil && qr.CPUSet == nil && qr.Threads == nil && qr.Journal == nil
```
Comment (mardy) [inline snap/quota/resources.go:208]: Would
```suggestion
	return *qr == Resources{}
```
work as well?

**Review by mardy — APPROVED** [review]
LGTM, thanks!

File: overlord/servicestate/quota_handlers.go [hunk 3]
```diff
@@ -416,8 +416,8 @@ func quotaRemove(st *state.State, action QuotaControlAction, allGrps map[string]
 		return nil, nil, false, fmt.Errorf("internal error, AddSnaps option cannot be used with remove action")
 	}
 
-	if action.ResourceLimits.Memory != nil {
-		return nil, nil, false, fmt.Errorf("internal error, MemoryLimit option cannot be used with remove action")
+	if !action.ResourceLimits.Unset() {
+		return nil, nil, false, fmt.Errorf("internal error, quota limit options cannot be used with remove action")
```
Comment (pedronis) [inline overlord/servicestate/quota_handlers.go:420]: nitpick: we tend to write "internal error: ..." with  a colon and not a comma. If we change it we should change the other ones here as well but as I said nitpick

**Pull request merged**